lib/searchyou/generator.rb

```diff
@@ -28,22 +34,37 @@ def generate(site)
       # Signal to the indexer that we're done adding content
       indexer.finish
 
-
     # Handle any exceptions gracefully
     rescue => e
       $stderr.puts "Searchyll: #{e.class.name} - #{e.message}"
+      $stderr.puts "Backtrace: #{e.backtrace.each{|l| puts l};nil}"
       raise(e)
     end
 
-    # Figure out the Elasticsearch URL, from an environment variable or the
-    # Jekyll site configuration. Raises an exception if none is found, so we
-    # can skip the indexing.
-    def elasticsearch_url(site)
-      ENV['BONSAI_URL'] || ENV['ELASTICSEARCH_URL'] ||
-      ((site.config||{})['elasticsearch']||{})['url'] ||
-      raise(ArgumentError, "No Elasticsearch URL present, skipping indexing")
-    end
+  end
 
+  # Create a configuration for the site
+  def self.configure(site)
+    self.configuration ||= Configuration.new(site)
   end
 
+  class Configuration
+    attr_accessor :url, :number_of_shards, :number_of_replicas, :index_name, :default_type
```

Comment:
Probably the only attribute should be `site`. The rest are custom getters that pull values from `site`.